### PR TITLE
Remove reset_gop_hash and update_gop_hash APIs

### PR DESCRIPTION
**Pull request by mithydolphin**
These APIs are no longer needed as recursive hash is not used.

### Describe your changes

Please include a summary of the change, a relevant motivation and context.

### Issue ticket number and link

- Fixes #(issue)

### Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have verified that the code builds perfectly fine on my local system
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have verified that my code follows the style already available in the repository
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: lib/src/signed_video_h26x_auth.c [hunk 1]
```diff
@@ -29,9 +29,9 @@
 #include "legacy_validation.h"
 #include "signed_video_authenticity.h"  // create_local_authenticity_report_if_needed()
 #include "signed_video_defines.h"  // svrc_t
-#include "signed_video_h26x_internal.h"  // gop_state_*(), update_gop_hash(), update_validation_flags()
+#include "signed_video_h26x_internal.h"  // gop_state_*() , update_validation_flags()
```
Comment (lusikamalo) [inline lib/src/signed_video_h26x_auth.c:32]: nit: remove space.

File: lib/src/signed_video_h26x_sign.c [hunk 2]
```diff
@@ -354,8 +354,8 @@ generate_sei_nalu(signed_video_t *self, uint8_t **payload, uint8_t **payload_sig
       memcpy(sign_data->hash, gop_info->gop_hash, hash_size);
     }
 
-    // Reset the gop_hash since we start a new GOP.
-    SV_THROW(reset_gop_hash(self));
+    // Reset the |num_nalus_in_gop_hash| since we start a new GOP.
+    self->gop_info->num_nalus_in_gop_hash = 0;
```
Comment (lusikamalo) [inline lib/src/signed_video_h26x_sign.c:358]: You already got "gop_info" from row 350.

**Review by lusikamalo — CHANGES_REQUESTED** [review]
Just some minor stuff.

**Review by bjornvolcker — APPROVED** [review]

File: lib/src/signed_video_h26x_common.c [hunk 3]
```diff
@@ -271,18 +271,6 @@ set_hash_list_size(gop_info_t *gop_info, size_t hash_list_size)
...
-svrc_t
-reset_gop_hash(signed_video_t *self)
-{
-  if (!self) return SV_INVALID_PARAMETER;
-
-  gop_info_t *gop_info = self->gop_info;
-  assert(gop_info);
-
-  gop_info->num_nalus_in_gop_hash = 0;
-  return openssl_hash_data(self->crypto_handle, &gop_info->gop_hash_init, 1, gop_info->gop_hash);
```
Comment (bjornvolcker) [inline lib/src/signed_video_h26x_common.c:283]: We should write a ticket to init the new gop hash computation with gop_hash_init.

**Review by bjornvolcker — APPROVED** [review]

**Review by lusikamalo — APPROVED** [review]
OK.

**Pull request merged**